ci/scripts/r_revdepcheck.sh

```diff
@@ -19,11 +19,40 @@
 set -ex
 
 : ${R_BIN:=R}
-
+: ${ARROW_REVDEP_WORKERS:=1)}
+: ${N_JOBS:=$(nproc)}
 source_dir=${1}/r
 
 # cpp building dependencies
-apt install -y cmake
+apt update -y -q && \
+apt install -y \
+  cmake \
+  libbrotli-dev \
+  libbz2-dev \
+  libc-ares-dev \
+  libcurl4-openssl-dev \
+  libgflags-dev \
+  libgoogle-glog-dev \
+  liblz4-dev \
+  libprotobuf-dev \
+  libprotoc-dev \
+  libradospp-dev \
+  libre2-dev \
+  libsnappy-dev \
+  libssl-dev \
+  libthrift-dev \
+  libutf8proc-dev \
+  libzstd-dev \
+  nlohmann-json3-dev \
+  pkg-config \
+  protobuf-compiler \
+  python3-dev \
+  python3-pip \
+  python3-rados \
+  rados-objclass-dev \
+  rapidjson-dev \
+  tzdata \
+  wget
```

Comment:
I wonder if we could actually "just" call something like https://github.com/apache/arrow/blob/master/dev/release/setup-ubuntu.sh and then we don't have to have all of these copied over? We have them in a few places already, though if my searching is right, most of them are in dockerfiles for various OSes

Reply:
While there is probably a better to do this, that script also installs stuff not needed here (llvm, openjdk...). So I'll keep it like this for now?

Reply:
Sure, maybe a TODO that mentions that we might want to consolidate it at some point?